packages/pull-signal/src/repositories/interop.repository.ts

```diff
@@ -16,14 +16,29 @@ export interface IInteropRepository {
       purposeId: string;
     }[]
   >;
+
+  findAdminActsInDelegationBy: (
+    eserviceId: string,
+    consumerId: string,
+    eserviceAllowedStates: string[],
+    purposeState: string,
+    agreementState: string,
+    delegationId: string
+  ) => Promise<
+    {
+      eserviceId: string;
+      agreementId: string;
+      purposeId: string;
+    }[]
```

Comment:
```suggestion
    AdministrativeAct[]
```
We can think about a sort of new abstraction